R/data-ddl-utils.R

```diff
@@ -0,0 +1,99 @@
+#' Function runs the `code`, masks the `code` and creates `teal_data` object.
+#' @param data (`teal_data`) object
+#' @param code (`language`) code to evaluate
+#' @param input (`list`) containing inputs to be used in the `code`
+#' @param input_mask (`list`) containing inputs to be masked in the `code`
+#'
+#' @return `teal_data` object
+#'
+#' @export
+eval_and_mask <- function(data,
+                          code,
+                          input = list(),
+                          input_mask = list()) {
```

Comment:
rename needed? Proposition: `env` and `env_mask`
This is so specific to the `shiny` and if you do https://github.com/insightsengineering/teal/pull/957/files#r1381599897 then there will be no link to the shiny at all. I feel this functionality is of general use.

Reply:
Good proposition, `env` and `env_mask` make sense as it is related directly to `substitute`.